flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierUnaligner.java

```diff
@@ -22,13 +22,10 @@
 import org.apache.flink.annotation.VisibleForTesting;
```

Comment:
Is the code in a working state before:
> Remove synchronization from CheckpointBarrierUnaligner.

commit? It looks like data are spilled in two places, right?